`TypeError`: '>' not supported between instances of 'NoneType' and 'str' in `viaduct.fetch_codes`

### Description
When running `test_fetch_codes` for viaducts with `update=True`, the method fails when trying to calculate the maximum last updated date. Some page fetches return `None` values, which cannot be compared with date strings in the `max()` function.

## Expected behaviour
The method should handle failed page fetches gracefully and calculate the maximum date from available data only.

## Error traceback
```
tests\test_viaduct.py:24 (TestViaducts.test_fetch_codes[True])
self = <tests.test_viaduct.TestViaducts object at ...>
vdct = <pyrcs.other_assets.viaduct.Viaducts object at ...>
update = True

    @pytest.mark.parametrize('update', [True, False])
    def test_fetch_codes(self, vdct, update):
>       vdct_codes = vdct.fetch_codes(update=update, verbose=True)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests\test_viaduct.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyrcs.other_assets.viaduct.Viaducts object at 0x00000296F5F51110>
page_no = None, update = True, dump_dir = None, verbose = True, kwargs = {}
codes_on_pages = [{'Last updated date': '2025-11-09', 'Page 1 (A-C)':                    Name  ... Spans
0              7 Arches  ...  ...  41m 72ch      
322   [unknown]             crosses Grand Union Canal  ...    41m 74ch      

[323 rows x 7 columns]}]

    def fetch_codes(self, page_no=None, update=False, dump_dir=None, verbose=False, **kwargs):
        ...
        ...
    
            viaducts_codes = {
                self.KEY: {next(iter(x)): next(iter(x.values())) for x in codes_on_pages},
                self.KEY_TO_LAST_UPDATED_DATE:
>                   max(next(itertools.islice(iter(x.values()), 1, 2)) for x in codes_on_pages),
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            }
E           TypeError: '>' not supported between instances of 'NoneType' and 'str'

pyrcs\other_assets\viaduct.py:225: TypeError
```

## Steps to reproduce
Run the specific failing tests:
```bash
pytest tests/test_viaduct.py::TestViaducts::test_fetch_codes -v
```

--- a/pyrcs/parser.py
+++ b/pyrcs/parser.py
@@ -73,7 +73,7 @@ def _prep_records(trs, ths, sep=' / '):
 
         for td_no, td in enumerate(tds):
             if td.find('td'):
-                text_ = td.find('a').contents + ["\t\t / "]
+                text_ = [''] if td.find('a') is None else td.find('a').contents + ["\t\t / "]
             else:
                 text_ = [_parse_other_tags_in_td_contents(x) for x in td.contents]
             # _move_element_to_end(text_, char='\t\t')

